ground-ingest/src/main/java/edu/berkeley/ground/ingest/FileMetadataSource.java

```diff
@@ -0,0 +1,60 @@
+package edu.berkeley.ground.ingest;
+
+import java.io.IOException;
+import java.util.List;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.google.common.collect.Lists;
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.State;
+import gobblin.configuration.WorkUnitState;
+import gobblin.source.extractor.Extractor;
+import gobblin.source.extractor.filebased.FileBasedHelperException;
+import gobblin.source.extractor.filebased.FileBasedSource;
+import gobblin.source.extractor.hadoop.HadoopFsHelper;
+import gobblin.util.HadoopUtils;
+
+
+/**
+ * A Source that extracts metadata from a file system.
+ */
+public class FileMetadataSource extends FileBasedSource<Schema, GenericRecord> {
```

Comment:
Yes this should be fine. Gobblin expects that users will extend / implement their own sources.